scipy/sparse/linalg/_isolve/lsmr.py

```diff
@@ -302,7 +302,7 @@ def lsmr(A, b, damp=0.0, atol=1e-6, btol=1e-6, conlim=1e8,
         return x, istop, itn, normr, normar, normA, condA, normx
 
     if normb == 0:
-        x = b
+        x[()] = 0
```

Comment:
Should we add the example regression tests from the matching issue? They look pretty small/simple enough?